src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -5143,6 +5144,13 @@ def _call_handler(self, node, parent):
 
         '''
         call = Call(parent=parent)
+        # TODO fparser/#447 For now `null()` is treated as a
+        # `Function_Reference` by fparser, instead of an
+        # `Intrinsic_Function_Reference` so we redirect to the correct
+        # handler for fparser2 tests to pass. This should be removed once
+        # fparser2 is fixed.
+        if str(node.items[0]).lower() == "null":
```

Comment:
Just to be on the safe side, please could you also check here that the call has no arguments. (I did some experiments and gfortran did let me define my own function called `null()`, albeit it hanged during execution.)

Reply:
This is done!